tests/functional/adapter/test_basic.py

```diff
@@ -40,6 +41,12 @@
 """
 
 
+@pytest.fixture(autouse=True)
+def slow_down_tests():
```

Comment:
We should have a comment indicating why this is the case.
And perhaps we choose a different name: e.g. `throttle_login_connections` or something that also indicates why this is being done.

Reply:
Fixed.